Xamarin.Forms.Core/Grid.cs

```diff
@@ -353,13 +355,25 @@ public void AddVertical(View view)
 				if (view == null)
 					throw new ArgumentNullException("view");
 
-				int lastRow = this.Any() ? this.Max(w => GetRow(w) + GetRowSpan(w) - 1) : -1;
-				lastRow = Math.Max(lastRow, Parent.RowDefinitions.Count - 1);
-				int lastCol = this.Any() ? this.Max(w => GetColumn(w) + GetColumnSpan(w) - 1) : -1;
-				lastCol = Math.Max(lastCol, Parent.ColumnDefinitions.Count - 1);
+				var rows = RowCount();
+				var columns = ColumnCount();
+
+				// if no columns, create a column
+				if (columns == 0)
+					columns++;
 
-				Add(view, 0, Math.Max(1, lastCol), lastRow + 1, lastRow + 2);
+				Add(view, 0, columns, rows, rows + 1);
 			}
+
+			private int RowCount() => Math.Max(
```

Comment:
This isn't getting called during the layout pass; This is really just a helper routine.